pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -361,7 +380,7 @@ func (a Adapter) execDevfile(commandsMap common.PushCommandsMap, componentExists
 	command, ok := commandsMap[versionsCommon.BuildCommandGroupType]
 	if ok {
 		compInfo.ContainerName = command.Exec.Component
-		err = exec.ExecuteDevfileBuildAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show, a.machineEventLogger)
+		err = exec.ExecuteDevfileBuildOrTestAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show, a.machineEventLogger)
```

Comment:
Maybe rename to something more simpler as even the init command implementation seems to also use this function.

Reply:
renamed to be `ExecuteDevfileCommands`
updated the comment for the function, `ExecuteDevfileCommands executes the devfile init, build and test command actions`